baobab/bnn_priors/models.py

```diff
@@ -0,0 +1,292 @@
+
+    References
+    ----------
+    .. [1] Arnouts, Stephane, et al. 
+    "The GALEX VIMOS-VLT Deep Survey* Measurement of the Evolution of the 1500 Å Luminosity Function." 
+    The Astrophysical Journal Letters 619.1 (2005): L43.
+
+    .. [2] Finkelstein, Steven L., et al. 
+    "The evolution of the galaxy rest-frame ultraviolet luminosity function over the first two billion years." 
+    The Astrophysical Journal 810.1 (2015): 71.
+
+    .. [3] Kawamata, Ryota, et al. 
+    "Size–Luminosity Relations and UV Luminosity Functions at z= 6–9 Simultaneously Derived from the Complete Hubble Frontier Fields Data." 
+    The Astrophysical Journal 855.1 (2018): 4.
+
+    Returns
+    -------
+    array-like
+        unnormalized function of the absolute magnitude at 1500A
+
+    """
+    #prefactor = np.log(10)*phi_star # just normalization
+    # Define redshift bins by right edge of bin
+    z_bins = np.array([0.2, 0.4, 0.6, 0.8, 1.2, 2.25, 3.4, 4.5, 5.5, 6.5, 7.5, np.inf])
+    alphas = np.array([-1.21, -1.19, -1.55, -1.60, -1.63, -1.49, -1.47, -1.56, -1.67, -2.02, -2.03, -2.36])
+    M_stars = np.array([-18.05, -18.38, -19.49, -19.84, -20.11, -20.33, -21.08, -20.73, -20.81, -21.13, -21.03, -20.89])
+    alpha = alphas[redshift < z_bins][0]
+    M_star = M_stars[redshift < z_bins][0]
+
+    # Note phi_star is ignored as normalization
+    # Schechter kernel
+    exponent = 10.0**(0.4*(M_star - M_grid))
+    density = np.exp(-exponent) * exponent**(alpha + 1.0)
+    return density
+
+def size_from_luminosity_and_redshift_relation(z, M_V):
+    """Sample the effective radius of Lyman break galaxies from the relation with luminosity and redshift
+
+    Parameters
+    ----------
+    z : float
+        redshift
+    M_V : float
+        V-band absolute magnitude
+
+    Note
+    ----
+    The relation and scatter agree with [1]_ and [2]_, which both show that size decreases
+    with higher redshift. They have been used in LensPop ([3]_).
+
+    References
+    ----------
+    .. [1] Mosleh, Moein, et al. 
+    "The evolution of mass-size relation for Lyman break galaxies from z= 1 to z= 7." 
+    The Astrophysical Journal Letters 756.1 (2012): L12.
+
+    .. [2] Huang, Kuang-Han, et al. 
+    "The bivariate size-luminosity relations for Lyman break galaxies at z∼ 4-5." 
+    The Astrophysical Journal 765.1 (2013): 68.
+
+    .. [3] Collett, Thomas E. 
+    "The population of galaxy–galaxy strong lenses in forthcoming optical imaging surveys." 
+    The Astrophysical Journal 811.1 (2015): 20.
+
+    Returns
+    -------
+    float
+        a sampled effective radius in kpc
+
+    """
+    log_R_eff = (M_V/-19.5)**-0.22 * ((1.0 + z)/5.0)**-1.2
+    scatter = np.random.randn()*0.3
+    log_R_eff += scatter
+    R_eff = 10.0**log_R_eff
+    return R_eff
+
+def axis_ratio_disklike(scale=0.3, truncate=0.2):
+    """Sample (one minus) the axis ratio of a disk-like galaxy from the Rayleigh distribution
+
```

Comment:
input parameters not specified in the doc string

Reply:
Updated